### PR TITLE
#131: Support storage node files

### PR DESCRIPTION
**Pull request by priscavdsluis**

### CODE REVIEW

File: hydrolib/core/io/storagenode/models.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        nodetype: Optional[str] = Field(
+            "(optional) Type of the node. Possible values are: "
+            + "inspection: inspection chamber, "
+            + "soakawayDrain: soakaway drain (infiltration), "
+            + "compartment: manhole compartment, "
+            + "unspecified: general storage node of unspecified type",
+            alias="nodeType",
+        )
+        nodeid: Optional[str] = Field(
+            "Connection node on which the storage nodeis located.", alias="nodeId"
```
Comment (arthurvd) [inline hydrolib/core/io/storagenode/models.py:72]: "node is"

File: hydrolib/core/io/storagenode/models.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            "Number of levels in storage area table.", alias="numLevels"
+        )
+        levels: Optional[str] = Field(
+            "Levels in storage area table [m].", alias="levels"
+        )
+        storagearea: Optional[str] = Field(
+            "Areas in storage area table [m2].", alias="storageArea"
+        )
+        interpolate: Optional[str] = Field(
+            "Interpolation type for storage area table either linear or block.",
```
Comment (arthurvd) [inline hydrolib/core/io/storagenode/models.py:108]: "Interpolation type for storage area table. Possible values: linear or block."

(I have updated the user manual already)

File: hydrolib/core/io/storagenode/models.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    @classmethod
+    def _validate_usetable(cls, values):
+
+        if values["usetable"]:
+            cls._validate_required_usetable_fields(
+                "numlevels", "levels", "storagearea", values=values, usetable=True
+            )
+
+            cls._validate_list_length(
+                "levels", "storagearea", length_field_name="numlevels", values=values
```
Comment (arthurvd) [inline hydrolib/core/io/storagenode/models.py:161]: Can you try and replace this with the existing utility, for example:
```
    _check_list_length = make_list_length_root_validator(
        "levels",
        "storagearea",
        length_name="numlevels",
        list_required_with_length=True,
    )
```
(last argument list_required_with_length comes from #179).

**Review by arthurvd — CHANGES_REQUESTED** [review]
Hi @priscavdsluis, nice fast work. See some of my minor remarks.
Additionally, could you add class API docs to all classes. See for example PR #166.

**Comment by arthurvd** [comment]
@priscavdsluis: I have also added your new functionality to the "support" list in Excel under commit 4ce7faf.

File: hydrolib/core/io/storagenode/models.py (hunk 3 above)
Comment (priscavdsluis) [reply hydrolib/core/io/storagenode/models.py:161]: Thanks, I will do that once it is in the master :)

Comment (priscavdsluis) [reply hydrolib/core/io/storagenode/models.py]: Done

**Comment by priscavdsluis** [comment]
@arthurvd I have added documentation to all classes.

**Review by arthurvd — APPROVED** [review]
I've added one minor commit, and I think this PR is now ready for merging.

**Comment by arthurvd** [comment]
Hey @priscavdsluis, another remark, just saw that coverage % reveals that `def _validate_list_length()` is now unused. Function can be removed I think?

**Comment by priscavdsluis** [comment]
@arthurvd , thanks for noticing. I have removed it!

**Pull request merged**